cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -80,7 +80,8 @@ public FactTable get(int resolution) {
     int ttl =  cConf.getInt(MetricsConstants.ConfigKeys.RETENTION_SECONDS + "." + resolution + ".seconds", -1);
 
     DatasetProperties.Builder props = DatasetProperties.builder();
-    if (ttl > 0) {
+    // don't add TTL for MAX_RESOLUTION table.
+    if (ttl > 0 && resolution != Integer.MAX_VALUE) {
```

Comment:
by default if TTL is not set , the TTL would be forever (INT_MAX) , if someone sets a lower TTL for the totals table, we want to ignore it and use TTL forever ( according to CDAP-1626). if we want to allow that to be configurable, then the issue and PR is not relevant right ?